Default text direction example and assertion

egekorkan opened the pull request:
fixes #1414

See https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/pull/1437.html#example-8 for an easier review


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/pull/1437.html" title="Last updated on Apr 5, 2022, 11:05 PM UTC (86e6841)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/1437/bafa33e...86e6841.html" title="Last updated on Apr 5, 2022, 11:05 PM UTC (86e6841)">Diff</a>

aphillips commented on index.html:
```diff
@@ -3022,6 +3008,49 @@ <h3>Human-Readable Metadata</h3>
+
+  <p>
+    Strings on the Web [[?STRING-META]] suggests 
+    both strong-first and language-based inferencing as means
```
`strong-first` should be `first-strong`

aphillips commented on index.html:
```diff
@@ -3022,6 +3008,49 @@ <h3>Human-Readable Metadata</h3>
 
+
+  <p>
+    Strings on the Web [[?STRING-META]] suggests 
```
We don't actually suggest both first-strong and inference from language tags. We actually recommend metadata falling back to first-strong. About language tags we say:

> This approach is only recommended as a workaround for situations that prevent the use of metadata.

Because you are using JSON-LD, you have metadata for the document as a whole and for individual string values (notably `title` and `description`). What you don't have is per-string direction in language maps (e.g. `titles` and `descriptions`) and you could recommend the use of the language tag for these.

I would thus tend to say:

```
Strings on the Web recommends the use of metadata to determine the <a>base 
direction</a> of string values. When metadata is not present, applications SHOULD 
use <a>first-strong detection</a>as a fallback. Language maps do not permit per-string
metadata: implementations MAY infer the base direction from the language tag (as
a better indicator than first-strong detection).
```



Note that the <a> links are found in `i18n-glossary` (you can add to your xrefs to pick up automagically)

aphillips commented on index.template.html:
```diff
@@ -1845,6 +1831,49 @@ <h3>Human-Readable Metadata</h3>
 
+
+  <p>
+    Strings on the Web [[?STRING-META]] suggests 
+    both strong-first and language-based inferencing as means
+    to determine the base text direction.
```
same comments

egekorkan commented:
@aphillips I have addressed your feedback. However, there is one small issue: we are using these definitions inside normative assertions but these definitions are informative in i18-glossary (which makes sense since it is just a glossary). Should I change something in the respec config?


**Important note for spec editors**: I have changed the respec config

aphillips reviewed (approved):
Looks good to me! Thanks!

sebastiankb commented:
many thanks!
